File: backend/internal/models/feature.go
package models

import "time"

// Feature représente un feature-flag géré depuis le back-office
type Feature struct {
	Key         string    `gorm:"type:varchar(255);primaryKey" json:"key"`
	Description string    `gorm:"type:text;not null" json:"description"`
	Enabled     bool      `gorm:"not null;default:false" json:"enabled"`
	CreatedAt   time.Time `gorm:"column:created_at;autoCreateTime" json:"created_at"`
	UpdatedAt   time.Time `gorm:"column:updated_at;autoUpdateTime" json:"updated_at"`
}

File: frontend/lib/src/models/feature.dart
class Feature {
  final String key;
  final String description;
  bool enabled;

  Feature({
    required this.key,
    required this.description,
    required this.enabled,
  });

  factory Feature.fromJson(Map<String, dynamic> json) => Feature(
    key: json['key'] as String,
    description: json['description'] as String,
    enabled: json['enabled'] as bool,
  );

  Map<String, dynamic> toJson() => {
    'key': key,
    'description': description,
    'enabled': enabled,
  };
}
